mixer/cmd/build.go

```diff
@@ -199,113 +213,163 @@ var buildFormatNewCmd = &cobra.Command{
 		if err != nil {
 			fail(err)
 		}
+		// Update mixer to build version +20, to build the bundles with +20 inside not +10
 		if err = b.UpdateMixVer(ver + 20); err != nil {
 			failf("Couldn't update Mix Version")
 		}
 
-		// Set format to format+1 so that the format file inserted into the
-		// update content is the new one
-		newFormat, err := strconv.Atoi(b.State.Mix.Format)
-		if err != nil {
-			fail(err)
-		}
-		newFormat++
-		if err = b.UpdateFormatVersion(strconv.Itoa(newFormat)); err != nil {
+		// Change the format internally just for building bundles so the right
+		// /usr/share/defaults/swupd/format is inserted; preserve old format
+		oldFormat := b.State.Mix.Format
+		b.State.Mix.Format = buildFlags.newFormat
+
+		// Build bundles normally. At this point the bundles to be deleted should still
+		// be part of the mixbundles list and the groups.ini
+		if err = buildBundles(b, buildFlags.noSigning, buildFlags.clean); err != nil {
 			fail(err)
 		}
-		setWorkers(b)
 
-		fmt.Println(" Backing up full groups.ini")
-		// Back up groups.ini in case we have deprecated bundles to delete
-		if err = b.CopyFullGroupsINI(); err != nil {
+		// Wipes the <bundle>-info files and replaces them with
```

Comment:
update this comment to say it is wiping the bundle-info files of the deprecated bundles only.

Reply:
I still think this comment should be updated, it currently makes it sound like we are wiping all the bundle-info files.